malariagen_data/util.py

```diff
@@ -204,10 +204,15 @@ def da_compress(indexer, data, axis):
     assert isinstance(indexer, da.Array)
     assert isinstance(axis, int)
     assert indexer.shape[0] == data.shape[axis]
+
+    # useful variables
     old_chunks = data.chunks
     axis_old_chunks = old_chunks[axis]
     axis_n_chunks = len(axis_old_chunks)
 
+    # ensure indexer and data are chunked in the same way
+    indexer = indexer.rechunk((axis_old_chunks,))
+
```

Comment:
I believe this is the key fix, rechunking the indexer here rather than later, although why this makes a difference is unclear.